utils/filters/filterHelpers.ts

```diff
@@ -0,0 +1,55 @@
+import { Prisma, RoleType } from '@prisma/client';
+import { canBecomeValidYear, isValidYear } from '../date/validations';
+import { canBecomeInteger } from '../numbers/validations';
+
+export function makeCompanyNameFilters(searchWords: string[]) {
+  // Do not want to return names that contain the year to prevent zero results.
```

Comment:
I don't understand this. Is it possible to explain?

Reply:
Based on the logic in the description, if we do something like `?searchQuery=2023`, this will return empty payload if there are no company names with `2023`

The alternative logic that I tried is too loose and returns too many unrelated applications.